subnet/deployment-generator/script/generate.sh

```diff
@@ -0,0 +1,46 @@
+#!/bin/bash
+VERSION_GENERATOR="generator-csc"
+VERSION_CSC="generator-update"
+VERSION_SUBNET="feature-puppeth-docker-2"
+
+current_dir="$(cd "$(dirname "$0")" && pwd)"
+echo 'checking for docker.env'
+if [ ! -f "docker.env" ]; then
+  echo 'docker.env not found'
+  exit 1
+fi
+
+if ! grep -q "CONFIG_PATH" "docker.env"; then
+  line="#current directory"$'\n'"CONFIG_PATH=$(pwd)"
+  echo "$line" | cat - "docker.env" > temp && mv temp "docker.env"
+  echo 'added CONFIG_PATH to docker.env'
+fi
+
+echo 'checking docker images'
+if [[ -z "$(docker images -q xinfinorg/subnet-generator:$VERSION_GENERATOR)" ]]; then # || echo "doesn't exist"
+  docker pull xinfinorg/subnet-generator:$VERSION_GENERATOR
+fi
+if [[ -z "$(docker images -q xinfinorg/csc:$VERSION_CSC)" ]]; then # || echo "doesn't exist"
+  docker pull xinfinorg/csc:$VERSION_CSC
+fi
+if [[ -z "$(docker images -q xinfinorg/xdcsubnets:$VERSION_SUBNET)" ]]; then # || echo "doesn't exist"
+  docker pull xinfinorg/xdcsubnets:$VERSION_SUBNET
+fi
```

Comment:
why you need to do these steps? If docker images not there, docker compose up will download it by default right?

Reply:
This is for speeding up when the images are already present. I found docker pull command takes a long time. 
It's not too big of an optimization. Does it look too error prone this way?

Reply:
no, all good then. If this is helping speeding up, then it's good to have.